File: src/main/java/problems/easy/GreatestCommonDivisorOfStrings/greatest_common_divisor_of_strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        print(f"str1 = {str1}, str2 = {str2}")
        if len(str1) < len(str2):
            if str1 in str2:
                return str1
        else:
            if str2 in str1:
                return str2
        return "-"

    def is_divisor_of(str1: str, str2: str) -> bool:
        return False

File: src/main/java/problems/easy/GreatestCommonDivisorOfStrings/GreatestCommonDivisorOfStrings.java
package problems.easy.GreatestCommonDivisorOfStrings;

import java.util.Arrays;

@SuppressWarnings("PMD")
class Solution {
  public String gcdOfStrings(String str1, String str2) {
    var result = "";
    final var maxDivisorLength =
        Arrays.stream(new int[] {str1.length(), str2.length()}).min().getAsInt();
    var currentDivisor = "";
    final var str1AsArray = str1.toCharArray();
    for (var i = 0; i < maxDivisorLength; i++) {
      currentDivisor += str1AsArray[i];
      if (isDivisorOf(currentDivisor, str1) && isDivisorOf(currentDivisor, str2))
        result = currentDivisor;
    }
    return result;
  }

  private Boolean isDivisorOf(final String divisor, final String data) {
    var result = false;
    if (divisor.length() > data.length()) return result;
    if (divisor.equals(data)) return true;
    final var listFromData = data.split(divisor);
    result = Arrays.asList(listFromData).stream().allMatch(item -> "".equals(item));
    return result;
  }
}
